src/neo/SmartContract/Native/TransactionState.cs

```diff
@@ -0,0 +1,25 @@
+using Neo.IO;
+using Neo.Network.P2P.Payloads;
+using Neo.VM;
+using Neo.VM.Types;
+
+namespace Neo.SmartContract.Native
+{
+    public class TransactionState : IInteroperable
```

Comment:
```csharp
    public class TransactionState : IInteroperable
    {
        public uint BlockIndex;
        public Transaction Transaction
        {
            get
            {
                if (_transaction != null) return _transaction;
                _transaction = _rawTransaction.AsSerializable<Transaction>();
                return _transaction;
            }
            set
            {
                _transaction = value;
                _rawTransaction = value.ToArray();
            }
        }

        private Transaction _transaction;
        private byte[] _rawTransaction;

        void IInteroperable.FromStackItem(StackItem stackItem)
        {
            Struct @struct = (Struct)stackItem;
            BlockIndex = (uint)@struct[0].GetInteger();
            _rawTransaction = @struct[1].GetSpan().ToArray();
            _transaction = null;
        }

        StackItem IInteroperable.ToStackItem(ReferenceCounter referenceCounter)
        {
            return new Struct(referenceCounter) { BlockIndex, _rawTransaction };
        }
    }
```
?

Reply:
What's the difference? The state is immutable. We won't change a stored transaction.

Reply:
Avoid duplicate serialization when calling `ToStackItem`

Reply:
Done.